src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/FileBinding.java

```diff
@@ -49,7 +49,9 @@
     }
 
     @Override public SingleEnvironment bindSingle(Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-        FileCredentials credentials = getCredentials(build);
+        if(workspace==null)
+        	return new SingleEnvironment("");
```

Comment:
:bug: the `workspace` parameter to `Binding.bindSingle` should be marked `@Nonnull` (ditto `MultiBinding.bind`) and this should be reverted.